src/scripts/file.ts

```diff
@@ -0,0 +1,53 @@
+import studentJSON from "./studentList";
+import {Student} from "~scripts/interfaces/studentInterface";
+import {RowEventHandler} from "~scripts/buttons";
+
+// type BUTTON_LABEL = "Refresh" | "Load Data";
+// let previousButtonValue: BUTTON_LABEL = "Refresh";
+const studentTableRows: RowEventHandler[] = [];
+
+function bootstrap() {
+    const buttonElement: HTMLButtonElement = <HTMLButtonElement>document.getElementById("totalButton");
+    buttonElement.addEventListener("click", switchButtonLabel);
+}
+
+function switchButtonLabel(this: HTMLButtonElement) {
+    if (this.innerText === 'Refresh') {
+        while (studentTableRows.length)
+            studentTableRows.shift()!.handleDelete();
+    } else {
+        this.innerText = 'Refresh';
+    }
+    showData();
+}
+
+function showData() {
+    const table: HTMLTableElement = <HTMLTableElement>document.getElementById("dataList");
+    const tbody: HTMLTableSectionElement = table.tBodies[0];
+    const template = <HTMLTemplateElement>document.getElementById("rowTmpl");
+
+
+    studentJSON.forEach((student: Student, i: number) => {
+        const rowTemplate = <HTMLTableRowElement>(
+            <DocumentFragment>template.content.cloneNode(true)
+        ).querySelector('tr');
+
+        rowTemplate.querySelectorAll("input")
+            .forEach((input) => {
+                input.style.display = "none";
+            });
+
+        rowTemplate.dataset.index = i.toString();
+
+        rowTemplate.cells[0].querySelector('label')!.innerText= tbody.rows.length.toString();
```

Comment:
Don't make null checks like this, as it  is not a check but a non-null assertion. This makes us responsible for the variable being not null. This could fail for edge cases, eg. all the elements got deleted from the dom and we somehow trigger this function again. So instead of checking it this way, make and if block handle the error gracefully.